client/src/components/Verify.js

```diff
@@ -150,6 +153,24 @@ class Verify extends Component {
               loading: false,
             });
             return null;
+          case 429:
+            this.setState({
+              loading: false,
+              message: i18n._t(
+                'MultiFactorAuthentication.TRY_AGAIN_ERROR',
+                fallbacks['MultiFactorAuthentication.TRY_AGAIN_ERROR']
+              )
+            });
+            return null;
+          case 500:
+            this.setState({
+              loading: false,
+              message: i18n._t(
+                'MultiFactorAuthentication.UNKNOWN_ERROR',
+                fallbacks['MultiFactorAuthentication.UNKNOWN_ERROR']
+              )
+            });
+            return null;
           default:
```

Comment:
I mentioned in the other PR that the "an unknown error" fallback would probably be suited to this `default:` case.

Reply:
I'm split on this.  I do think it's a good idea, though I'm also concerned that we may cause a regression by doing this because I did notice there was at least on 4** code that came back from the server with a message (maybe it was on failed verification?).  Probably the wrong status code was used there.  By limiting this to 500, at least can at least be confident we're not breaking any functionality with this pull-request, and we'll fix the linked issue which was specifically for 500.

Reply:
Currently, correct me if I'm wrong, it never resets the spinner in any case where we don't get a `422`, `200`, or `202` right?

I think the catch all with at least an indication that something is wrong is preferable to never seeing any update.

Reply:
Not quite, the current logic will stop the spinner if valid json is returned.  If non-valid json, such a blank on exception on a live environment, then the spinner remains. I think the spinner stops as a result of `this.setState({loading: false})`

```js
          default:
        }
        return response.json();
      })
      .then(result => {
        if (result) {
          this.setState({
            loading: false,
            ...result,
          });
        }
      });
```

>> I think the catch all with at least an indication that something is wrong is preferable to never seeing any update.

Agree, though I'm worried we might break something in the process.

If you can show me some way to be more confident that the catch all won't break anything ([a valid 400 response?](https://github.com/silverstripe/silverstripe-mfa/blob/4/client/src/components/tests/Verify-test.js#L293)), then I'll change the `500` to `default`, otherwise I think we should keep as

Reply:
We're already catching the most common and only valid responses (`200` and `202`). Everything else is an error of some sort. "A valid 400 response" is still not a valid response, it's a "you screwed up" response to the client. The client can choose to handle that 400 _error_, and I'm saying all _error_'s that are unhandled specifically by that switch statement can at least still give feedback in the form of "well we didn't expect that one, but there was an error btw".

If you want, I'll be happy with the `default` case checking that the status code starts with a `5` before responding with "an unknown error" if you're really concerned.

Reply:
I've updated to treat 5** as a catch-all